### ISSUE TITLE
Field holding onto previous props?

### ISSUE THREAD

**Issue by Aaronius** [erikras/redux-form#1785]
This is demonstrated in [this JSBin](https://jsbin.com/reparab/3/edit?html,js,output). I have a toggle button that, when clicked, should essentially change the `Field` between:

```
<Field
  name="foo"
  component="input"
  type="text"
/>
```

and 

```
<Field
  name="foo"
  component="textarea"
/>
```

When clicking the toggle button, you'll notice it doesn't immediately switch between an input and a textarea. However, if you then click on the input or textarea, it will then usually switch to what it should be. It's like props aren't being propagated correctly until the following render cycle.

If I either never set `type` or I always set `type` (to either `"text"` or `""`) then the input/textarea will immediately toggle when clicking the toggle button as expected.

Anyway, something funky is going on. I apologize if it turns out to just be a fundamental misunderstanding of React props or something similar.

**Comment by Aaronius** [comment]
I noticed this: https://github.com/erikras/redux-form/blob/89e183f6393980860c4f3de88a6ab15d08ca24f6/src/ConnectedField.js#L23-L25

Shouldn't it be more like:

```
return nextPropsKeys.length !== thisPropsKeys.length || nextPropsKeys.some(prop => {
  return !~propsToNotUpdateFor.indexOf(prop) && !deepEqual(this.props[ prop ], nextProps[ prop ])
})
```

**Comment by Aaronius** [comment]
FWIW, that seems to fix the issue.

**Comment by andi5** [comment]
What would be the proper way to ask for a release containing that fix? :-)

**Comment by Aaronius** [comment]
I could really use a build as well.

**Comment by erikras** [comment]
Published in [`v6.1.0`](https://github.com/erikras/redux-form/releases/tag/v6.1.0).
